gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java

```diff
@@ -589,6 +589,24 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
         });
   }
 
+  public FSDataInputStream open(FileStatus status) throws IOException {
+    logger.atFine().log("openWithStatus(%s)", status);
+
+    GoogleHadoopFileStatus fileStatus = (GoogleHadoopFileStatus) status;
```

Comment:
Wouldn't it be casting issue?

Reply:
Caller will have to handle the exception.

Reply:
It would be better to validate and  throw IllegalArgument instead of throwing type cast exception from API?

Reply:
Done